### ISSUE TITLE
0 value salesInvoices

### ISSUE THREAD

**Issue by shedges7** [Sage/SData-2.0#178]
Since upgrading Sage to V28 all the salesInvoices and salesCredits we are uploading using sData are showing with a 0.00 value in Sage. 
We haven't changed anything in our sData app, and using V27.2 the invoices & credits were being written as expected. 

I've used fiddler to check that the XML contains netTotals on the lines and can see the values in the XML. Any idea's why the lines, and total, is 0.00 when viewing in Sage?

**Comment by darroncockram** [comment]
Nothing has changed with the Sdata adapter so this _might_ be a bug on the Accounts side of things. Can you provide an example payload that is resulting in zero values please and I'll see if I can replicate.

**Comment by shedges7** [comment]
Hi Darron, attached is the XML for a request that resulted in a zero value invoice, I have removed some of the identifiable info and repalced with { }

[sData 0 value Invoice Request.txt](https://github.com/Sage/SData-2.0/files/9020746/sData.0.value.Invoice.Request.txt)

Thank you

**Comment by darroncockram** [comment]
Managed to replicate the issue based on your example payload so I compared this to the payload that results from my code sample for posting an invoice (https://github.com/darroncockram/sdatasamples/tree/master/CreateSalesInvoice), as that does not suffer from the same issue. From that I determined that the key difference is that I'm setting the actualPrice field (which maps to the unit price field in Accounts). If I simply add <actualPrice>456.45</actualPrice> to the salesInvoiceLineItem in your example payload it works correctly.

I ran a quick test against v27 and the payload works both with and without actualPrice. I've not been able to pin down where the change in behaviour with v28 has come from, or it is intentional, but simply setting actualPrice seems to do the trick and is compatible with older versions, so that would be the best bet for now.
